modules/Bio/EnsEMBL/Hive/DBSQL/AnalysisJobAdaptor.pm

```diff
@@ -587,6 +587,59 @@ sub reset_or_grab_job_by_dbID {
     return $job;
 }
 
+=head2 reset_job_by_input_id_and_sync
+
+  Arg [1]: string $input_id pattern
+  Arg [2]: array $analsyes_object_array
+  Example:
+    my $job = $self->{'dba'}->get_AnalysisJobAdaptor->reset_job_by_input_id_and_sync($input_id_pattern, $analsyes_object_array);
+  Description:
+    Reset jobs for the specified $input_id pattern and $analsyis_object_array. $input_id can be a wildcard argument.
+  Returntype : none
+  Exceptions :
+  Caller     : beekeeper.pl
+
+=cut
+
+sub reset_job_by_input_id_and_sync {
+    my ($self, $input_id_pattern, @analyses_list) = @_;
+
+    #Convert MySQL wildcard pattern to Perl wildcard pattern
+    if ($input_id_pattern =~ /\A[^%]/) {
+        $input_id_pattern = '^'.$input_id_pattern;
+    }
+    if ($input_id_pattern =~ /[^%]\z/) {
+        $input_id_pattern = $input_id_pattern.'$';
+    }
+    $input_id_pattern =~ s/\%/\.*/g;
+
+    my $analysis;
+    my $jobs = $self->fetch_all_by_analysis_id_status(@analyses_list);
+    foreach my $job (@$jobs) {
+         my $input_id = $job->input_id;
+         $input_id_pattern = qr/$input_id_pattern/;
+         if ($input_id =~ /$input_id_pattern/) {
+                my $job_status = $job->status();
+                my $job_info = $job->toString;
+                if($job_status =~ /CLAIMED|SEMAPHORED/) {
+                       die "$job_info is $job_status, cannot reset";
```

Comment:
I'm not sure the code should _die_ here, maybe simply skip the job ?
This would definitely be the correct behaviour for "reset by job_id", i.e. "Can you reset job_id 1234 for me ? No, I can't, it's CLAIMED".
But what do you think the question is here ?
1. "Can you reset _all_ the jobs that have `species => human` for me ?" -> Should die if at least 1 job can't be reset, probably including jobs that are running
1. "Can you reset all the _resettable_ jobs that have `species => human` for me ?" Should silently skip the jobs that are already running / claimed or can't be reset.